base/strings/replace.jl

```diff
@@ -0,0 +1,316 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+# This file depends on base/regex.jl and base/show.jl
+# and needs to follow these in base/sysimg.jl
+
+"""
+    Module to implement replace method with multiple pairs of patterns-replacements
+"""
+module ReplaceImpl
+
+export replace_code
+import Base: Callable, SubstitutionString
+
+# types of input patterns
+const TIF = Union{Char,AbstractString,Regex,Callable,Tuple{Vararg{Char}},AbstractVector{Char},Set{Char}}
+# types of input replacements
+#const TIS = Union{Char,AbstractString,Callable,SubstitutionString}
+const TIS = Any
+
+# types input patterns are converted to
+const TF = Union{Regex,String,Callable}
+# types output patterns are converted to
+const TS = Union{Char,String,Callable,SubstitutionString}
+
+# from reflection.jl::hasmethod
+function type_hasmethod(@nospecialize(ft::DataType), @nospecialize(t), world=typemax(UInt))
+    t = Tuple{ft, t...}
+    ccall(:jl_method_exists, Cint, (Any, Any, UInt), ft.name.mt, t, world) != 0
```

Comment:
It's illegal to call this from a generated function: it's simply never permissible to introspect the method tables.